### PR TITLE
feat(node): implement starknet json rpc query mutate

### LINKED ISSUE #828: Starknet Integration - starknet json rpc query method
Following Near integration example, add Starknet json rpc integration for query method.

### PR DESCRIPTION
**Pull request by frdomovic**
# feat(node): implement starknet json rpc query mutate

## Summary
Added Network method mutate for calling Starknet smart contract.

Fixes #828

### CODE REVIEW

File: crates/context/config/src/client/starknet.rs [hunk 1]
```diff
@@ -165,4 +190,88 @@ impl Network {
             }),
         }
     }
+
+    async fn mutate(&self, contract_id: &str, method: &str, args: Vec<u8>) {
```
Comment (miraclx) [inline crates/context/config/src/client/starknet.rs]: return a `Result`

File: crates/context/config/src/client/starknet.rs [hunk 2]
```diff
@@ -165,4 +190,88 @@ impl Network {
...
+        let sender_address: Felt = Felt::from_str(&self.account_id)
+            .unwrap_or_else(|_| panic!("Failed to convert sender address to felt type"));
```
Comment (miraclx) [inline crates/context/config/src/client/starknet.rs]: no panic please

File: crates/context/config/src/client/starknet.rs [hunk 3]
```diff
@@ -120,6 +121,30 @@ impl Transport for StarknetTransport<'_> {
...
+    contract_address: Felt,
+    entry_point_selector: Felt,
+    calldata: Vec<Felt>,
+) -> Felt {
+    let elements: Vec<Felt> = vec![sender_address, contract_address, entry_point_selector]
+        .into_iter()
+        .chain(calldata.into_iter())
+        .collect();
+
+    poseidon_hash_many(&elements)
```
Comment (miraclx) [inline crates/context/config/src/client/starknet.rs]: no need to allocate

```suggestion
    let elements = [sender_address, contract_address, entry_point_selector]
        .iter()
        .chain(calldata.iter());

    poseidon_hash_many(elements)
```

File: crates/context/config/src/client/starknet.rs [hunk 4]
```diff
@@ -165,4 +190,88 @@ impl Network {
...
+                    Felt::from_bytes_be(&padded_chunk)
+                })
+                .collect()
+        };
+
+        let transaction_hash = compute_transaction_hash(
+            sender_address,
+            contract_id,
+            entry_point_selector,
+            calldata.clone(),
```
Comment (miraclx) [inline crates/context/config/src/client/starknet.rs]: you can save this extra allocation by: ..._

```suggestion
            &calldata,
```

File: crates/context/config/src/client/starknet.rs [hunk 5]
```diff
@@ -120,6 +121,30 @@ impl Transport for StarknetTransport<'_> {
     }
 }
 
+fn compute_transaction_hash(
+    sender_address: Felt,
+    contract_address: Felt,
+    entry_point_selector: Felt,
+    calldata: Vec<Felt>,
```
Comment (miraclx) [inline crates/context/config/src/client/starknet.rs]: _...

```suggestion
    calldata: &[Felt],
```

File: crates/context/config/src/client/starknet.rs (hunk 1 above)
Comment (frdomovic) [reply crates/context/config/src/client/starknet.rs]: updated

File: crates/context/config/src/client/starknet.rs (hunk 2 above)
Comment (frdomovic) [reply crates/context/config/src/client/starknet.rs]: updated

File: crates/context/config/src/client/starknet.rs (hunk 3 above)
Comment (frdomovic) [reply crates/context/config/src/client/starknet.rs]: updated

File: crates/context/config/src/client/starknet.rs (hunk 4 above)
Comment (frdomovic) [reply crates/context/config/src/client/starknet.rs]: updated

File: crates/context/config/src/client/starknet.rs (hunk 5 above)
Comment (frdomovic) [reply crates/context/config/src/client/starknet.rs]: updated

File: crates/context/config/src/client/starknet.rs [hunk 6]
```diff
@@ -158,6 +167,30 @@ impl Transport for StarknetTransport<'_> {
...
+    contract_address: Felt,
+    entry_point_selector: Felt,
+    calldata: &[Felt],
+) -> Felt {
+    let elements: Vec<&Felt> = vec![&sender_address, &contract_address, &entry_point_selector]
+        .into_iter()
+        .chain(calldata.iter())
+        .collect();
+
+    poseidon_hash_many(elements)
```
Comment (miraclx) [inline crates/context/config/src/client/starknet.rs]: you shouldn't need to allocate - no `vec!` or `collect` necessary

```suggestion
    let elements = [sender_address, contract_address, entry_point_selector]
        .iter()
        .chain(calldata.iter());

    poseidon_hash_many(elements)
```

Comment (frdomovic) [reply crates/context/config/src/client/starknet.rs]: Updated

File: crates/context/config/src/client/starknet.rs [hunk 7]
```diff
@@ -158,6 +167,31 @@ impl Transport for StarknetTransport<'_> {
...
 
+fn compute_transaction_hash(
+    sender_address: Felt,
+    contract_address: Felt,
+    entry_point_selector: Felt,
+    calldata: &[Felt],
+) -> Felt {
+    // let elements: Vec<&Felt> = vec![&sender_address, &contract_address, &entry_point_selector]
+    //     .iter()
+    //     .chain(calldata.iter());
```
Comment (miraclx) [inline crates/context/config/src/client/starknet.rs]: ^

**Review by miraclx — APPROVED** [review]

File: crates/context/config/src/client/starknet.rs (hunk 7 above)
Comment (frdomovic) [reply crates/context/config/src/client/starknet.rs]: removed

**Pull request merged**